pkg/services/govmomi/service.go

```diff
@@ -326,6 +327,9 @@ func createIPAddressClaim(ctx *context.VMContext, ipAddrClaimName string, poolRe
 				},
 			},
 			Finalizers: []string{infrav1.IPAddressClaimFinalizer},
+			Annotations: map[string]string{
+				clusterv1.ClusterLabelName: ctx.VSphereVM.ObjectMeta.Annotations[clusterv1.ClusterLabelName],
+			},
```

Comment:
This needs to be moved within https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/6c50469d4c890ad32e39bdd81c3c0aa932e16f97/controllers/vspherevm_ipaddress_reconciler.go#L137-L153